api/api.go

```diff
@@ -22,20 +22,24 @@ var (
 
 const (
 	// Resource name regex
+	// Gorilla mux encloses the passed pattern with '^' and '$'. So we need to do some tricks
+	// to have mux eventually build a query regex which matches empty or word string (`^$|[\w]+`)
 	regex = "[a-zA-Z_0-9-]+"
+	qregx = "$|" + regex
 	// Router URL variable definition
-	nwName = "{" + urlNwName + ":" + regex + "}"
-	nwID   = "{" + urlNwID + ":" + regex + "}"
-	nwPID  = "{" + urlNwPID + ":" + regex + "}"
-	epName = "{" + urlEpName + ":" + regex + "}"
-	epID   = "{" + urlEpID + ":" + regex + "}"
-	epPID  = "{" + urlEpPID + ":" + regex + "}"
-	cnID   = "{" + urlCnID + ":" + regex + "}"
-
-	// Though this name can be anything, in order to support default network,
-	// we will keep it as name
-	urlNwName = "name"
-	// Internal URL variable name, they can be anything
+	nwName   = "{" + urlNwName + ":" + regex + "}"
+	nwNameQr = "{" + urlNwName + ":" + qregx + "}"
```

Comment:
Yes, nwName is used as URL parameter, it matches on valid nw names only.
nwNameQr is for query parameters and matches on empty string and nw names.

We cannot use nwNameQr for both url and query prms because it would not be right to match an empty string for URL parameter.